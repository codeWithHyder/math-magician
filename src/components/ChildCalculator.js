import React from 'react';
import PropTypes from 'prop-types';

function ChildCalculator({ label, onClick }) {
  const isWide = label === '0';
  return (

    <button
      type="button"
      className={`calculator-button ${isWide ? 'wide' : ''}`}
      onClick={onClick}
    >
      {label}
    </button>

  );
}

ChildCalculator.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default ChildCalculator;
