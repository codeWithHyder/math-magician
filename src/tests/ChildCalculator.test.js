import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ChildCalculator from '../components/ChildCalculator';

it('test button render', () => {
  const obj = ['1', '2', '3', '+'];
  const cName = 'calculator-button ';
  const tree = renderer.create(<ChildCalculator label={obj[0]} handleClick={() => handleClick(obj[0])} />).toJSON();

  render(<ChildCalculator label={obj[0]} handleClick={() => handleClick(obj[0])} />);
  const button = screen.getByRole('button');
  expect(button.innerHTML).toBe(obj[0]);

  expect(button.className).toBe(cName);
  expect(button.id).toBeFalsy();
  expect(tree).toMatchSnapshot();
});
