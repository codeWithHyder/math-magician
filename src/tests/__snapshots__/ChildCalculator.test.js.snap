// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button render 1`] = `
<button
  className="calculator-button "
  type="button"
>
  1
</button>
`;
